languages/js/src/Oso.ts

```diff
@@ -300,27 +321,25 @@ export class Oso<
    * @param resourceCls Object type.
    * @returns An array of authorized resources.
    */
-  async authorizedResources(
-    actr: any,
-    actn: any,
-    resourceCls: any
-  ): Promise<any[]> {
-    const query = await this.authorizedQuery(actr, actn, resourceCls);
-    return !query
-      ? []
-      : this.getHost().types.get(resourceCls)!.execQuery!(query);
+  async authorizedResources<T>(
+    actor: Actor,
+    action: Action,
+    resourceCls: Class<T>
```

Comment:
Wonder if there's a way to tie this `Class<T>` to the `Resource` type param on the `Oso` class

Reply:
Also as an aside our `Class` type seems very similar to TypeScript's built-in [`InstanceType<Type>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#instancetypetype)

Reply:
> Wonder if there's a way to tie this `Class<T>` to the `Resource` type param on the `Oso` class

turns out there is: [`8d49c50` (#1211)](https://github.com/osohq/oso/pull/1211/commits/8d49c505a9d6c96f586d98f977b0836cc46e27d3#diff-2178a05ab0077a949c52ac5f9d679b82d59967f2a2536708262b202d53dffee6R324)